snapshot-labs/envelop
# feat: Ignore flagged spaces and proposals

## 🧿 Current issues / What's wrong ?

Current emails do not filter out flagged spaces and proposals

## 💊 Fixes / Solution

Fix #74 

Use sidekick moderation list to filter out flagged spaces and proposals from all email types

## 🚧 Changes

- Filter out flagged spaces and proposals from summary emails
- Filter out flagged proposals from new/closed proposal emails

## 🛠️ Tests

- Go to `https://sh5.co/api/moderation` and join some spaces that are flagged (`flaggedSpaces` key) on snapshot, with your address
- Go to `/preview/summary?id=ADDRESS` (replace ADDRESS by your wallet address). The preview summary email should not contains any spaces nor proposals from the flagged spaces you just joined

Assuming you have at least on subscriber subscribed to all email types in your dev DB:

- Use `yarn ts-node scripts/trigger-webhook.ts proposal/created [ID]` (replace ID by a random flagged proposal ID from the sh5.co list). It should prints `Queued 0 email job`.
